src/app/portlets/dot-edit-page/dot-template-containers-cache.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { DotContainer } from '../../shared/models/container/dot-container.model';
+
+/**
+ * Save into cache the containers used by the current template
+ */
+@Injectable()
+export class TemplateContainersCacheService {
+    private containers: { [key: string]: {container: DotContainer}};
+
+    setContainers(containers: { [key: string]: {container: DotContainer}}): void {
+        this.containers = containers;
+    }
+
+    getContainer(containerId: string): DotContainer {
+        return this.containers[containerId] ? this.containers[containerId].container :  null;
+    }
+}
```

Comment:
Why don’t call the method just set and get? Because the name of the instance will be explicit enough to understand that we are getting and setting containers